cmake/Dependencies.cmake

```diff
@@ -0,0 +1,22 @@
+include(FetchContent)
+
+FetchContent_Declare(
+    clap
+    GIT_REPOSITORY https://github.com/free-audio/clap.git
+    GIT_TAG main
+    SYSTEM
+    # 'FIND_PACKAGE_ARGS' will skip download if
+    # the target is already available in the system
+    FIND_PACKAGE_ARGS NAMES clap
+)
+
+FetchContent_Declare(
+    clap-helpers
+    GIT_REPOSITORY https://github.com/free-audio/clap-helpers.git
+    GIT_TAG main
+    SYSTEM
+    FIND_PACKAGE_ARGS NAMES clap-helpers
+)
+
+
+FetchContent_MakeAvailable(clap clap-helpers)
```

Comment:
This is the interesting bit: with `FIND_PACKAGE_ARGS`, if `find_package` is called and the package is already _somewhere_ in the system, then `FetchContent` won't download the dependency.

I like it because it's very non intrusive: you can get your dependencies however it works for you (distro package manager, conan, homebrew, some custom cmake scripts, whatever) and, if none of that is available, `FetchContent` is used as a fallback to make things JustWork.

Reply:
I don't see the benefit.
clap-host and clap-plugin aren't meant to be submodules of other projects.
They need a precise version of clap and clap-helpers to be built with.

Reply:
Sorry, I think I didn't make myself clear: this is about adding `clap` and `clap-helpers` as dependencies of `clap-host` using native cmake features only.

We can tie them to specific versions (that need to be git tags or branches in the fetched repos) using the `GIT_TAG` property, I'll update the PR.

Reply:
> Sorry, I think I didn't make myself clear: this is about adding `clap` and `clap-helpers` as dependencies of `clap-host` using native cmake features only. `FetchContent` can be thought of as a cmake native `CPM`, even if a little more lower level.
> 
> We can tie them to specific versions (that need to be git tags or branches in the fetched repos) using the `GIT_TAG` property, I'll update the PR.

But having them as submodules does almost the same (tied to a version, can be fetched and so on.). And they are both not very large repos so it's no problem to have them on the drive twice. So I think it's fine to have them as submodules

Reply:
the biggest advantage is that, if you have a system wide installation of `clap` and `clap-helpers`, then `FetchContent` will use that instead of downloading their sources (fixing issues like https://github.com/free-audio/clap-host/issues/29).

Same would work if you call, say, `cmake .. -DCMAKE_PREFIX_PATH=/some/folder/with/clap/and/clap-helpers/installed`: if cmake can find the packages ready to use it'll use them, and fallback to downloading their sources only if needed (as opposed to the git submodules in which they _always_ have to be downloaded).

Reply:
also, I'm probably not the only one who constantly forgets the `--recurse-submodules` on `git clone` and/or `git submodule update --init`, but that's on me 😬